### ISSUE TITLE
Error during code generation when using DDLDatabase on a table with a MySQL enum type

### ISSUE THREAD

**Issue by quen2404** [jOOQ/jOOQ#7813]
Hello everyone, 
as asked by @lukaseder in  https://github.com/jOOQ/jOOQ/issues/5874, i create this bug report :) 

### Expected behavior and actual behavior:
I have in my sql schema an sql enum type for one field: 
```sql
CREATE TABLE `operator` (
  `uid` bigint(20) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
  `creation_timestamp` datetime DEFAULT CURRENT_TIMESTAMP,
  `state` enum('ACTIVE','SUSPENDED','DEACTIVATED') NOT NULL,
  `state_timestamp` datetime NOT NULL,
  `external_id` varchar(255) NOT NULL UNIQUE,
  `external_href` varchar(512) NOT NULL
);
```
The generation failed with this message : 
```
/org/jooq/impl/GeneratedEnum23734746.java:2: error: cannot find symbol
enum GeneratedEnum23734746 implements org.jooq.EnumType {
                                              ^
  symbol:   class EnumType
  location: package org.jooq
/org/jooq/impl/GeneratedEnum23734746.java:4: error: method does not override or implement a method from a supertype
  @Override
  ^
/org/jooq/impl/GeneratedEnum23734746.java:8: error: method does not override or implement a method from a supertype
  @Override
  ^
3 errors
```
It works correctly when i replace state column by this: 
```sql
`state` varchar(64) NOT NULL,
```

### Steps to reproduce the problem (include example code if possible):
I created this project to reproduce my error: https://github.com/quen2404/test-jooq

### Versions:

- jOOQ: 3.11.4
- Java: 10
- Database (include vendor): N/A
- OS: Windows
- JDBC Driver (include name if inofficial driver): N/A

**Comment by lukaseder** [comment]
Thank you very much for your report. I can reproduce this problem. Will investigate

**Comment by lukaseder** [comment]
It looks as though we need to pass some additional class path information to the compiler, depending on how we're calling the compiler (Maven, Java version). I'm not 100% sure what caused this regression, knowing that this has worked in the past on some environment, but I can definitely reproduce it now.

The fix has been applied to jOOR and will be moved to jOOQ as well: https://github.com/jOOQ/jOOR/issues/65

**Comment by quen2404** [comment]
Ok thank you for your quick response. So i have to wait JooQ 3.12 ? 
Any idea about the release date ?

**Comment by lukaseder** [comment]
I haven't finished my work here yet, stay tuned :-)

**Comment by lukaseder** [comment]
This is now fixed for jOOQ 3.12. Backport scheduled for 3.11.5 (#7815) (due by mid September)
